### PR TITLE
docs: dev instructions for GN build

### PR DESCRIPTION
**Pull request by nornagon**

### CODE REVIEW

File: docs/development/build-instructions-gn.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+$ gclient sync --with_branch_heads --with_tags
+# This will take a while, go get a coffee.
+```
+
+## Building
+
+```sh
+$ cd src
+$ export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
+$ gn gen out/Default --args='root_extra_deps=["//electron"] is_electron_build=true is_component_build=true use_jumbo_build=true'
```
Comment (deepak1556) [inline docs/development/build-instructions-gn.md]: Can you also add the other required flags from https://github.com/electron/libchromiumcontent/blob/master/chromiumcontent/args/shared_library.gn for example `v8_*` which is required by node. We seem to checkout nacl, in that case should we add `enable_nacl=false` ?

Comment (nornagon) [reply docs/development/build-instructions-gn.md]: chrome 63 doesn't have a way to disable checking out nacl. As far as I can tell, `enable_nacl` doesn't actually affect anything we depend on (I grepped and looked at all the places it's referenced, and also I don't have any nacl-related build artifacts in my `out/Default` directory, despite the fact that I haven't set `enable_nacl = false`, so I conclude that it's not necessary). I'll add the `v8_*`, though it seems like v8's been working OK without them... maybe it's subtly broken though and I just haven't noticed!

I'm also looking into better ways to specify these args than passing them all in the command line copy/pasted from the docs.

**Comment by nornagon** [comment]
@electron/docs could we get an lgtm?

File: docs/development/build-instructions-gn.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+$ ./out/Default/electron
+```
+
+## Tests
+
+To run the tests, you'll first need to build the test modules against the
+same version of node.js that was built as part of the build process.
+
+```sh
+$ (cd electron/spec && npm i --nodedir=../../third_party/electron_node)
```
Comment (zeke) [inline docs/development/build-instructions-gn.md]: why the parens here?

Comment (nornagon) [reply docs/development/build-instructions-gn.md]: Parens create a subshell, making it so the `cd` doesn't effect the current shell.

```sh
$ pwd
/Users/nornagon
$ (cd work; pwd)
/Users/nornagon/work
$ pwd
/Users/nornagon
```

**Pull request merged**